src/dsp_tools/commands/ingest_xmlupload/bulk_ingest_client.py

```diff
@@ -82,3 +84,36 @@ def upload_file(
         print(f"{datetime.now()}: {msg}")
         logger.info(msg)
         return None
+
+    def kick_off_ingest(self) -> None:
+        """Start the ingest process on the server."""
+        url = f"{self.dsp_ingest_url}/projects/{self.shortcode}/bulk-ingest"
+        res = self.session.post(url, timeout=5)
+        if res.status_code == STATUS_CONFLICT:
+            msg = f"Ingest process on the server {self.dsp_ingest_url} is already running. Wait until it completes..."
+            print(msg)
+            logger.info(msg)
+        if res.json().get("id") != self.shortcode:
+            raise UserError("Failed to kick off the ingest process.")
```

Comment:
I'm not sure under what scenario this would happen, but is it really a User/Input problem when the response from the server does not contain the ID provided?

But am I seeing it right that you do not check for any other status code (404, 500, etc.)? What happens if you kick off an ingest of an unknown shortcode? shouldn't that result in 404? And if so, I think `.json()` will not give a user-friendly message...

Reply:
Concerning the UserError: The underlying problem is that our exception hierarchy lacks a clear concept. This point is already noted on Notion, and I have added it to the Topics to discuss: https://www.notion.so/dasch-swiss/New-error-hierarchy-45fe23b41f254555843d7228552ac18a.

Concerning the other cases: Thanks for pointing it out. I have now handled all conceivable cases (https://github.com/dasch-swiss/dsp-tools/commit/97cea18b4600d6865731ac44f84ea26492bf21bf), and written unit tests for them (https://github.com/dasch-swiss/dsp-tools/commit/b937331a432542c820f2ea34633cd8ef8730dabd).